grails-app/services/com/recomdata/transmart/data/export/HighDimExportService.groovy

```diff
@@ -43,17 +42,20 @@ class HighDimExportService {
      *        factory, e.g., <code>names</code> for the <code>genes</code> filter type. 
      */
     def DataConstraint createFilterConstraints(HighDimensionDataTypeResource dataTypeResource, List<Map> filters) {
-        def dataConstraints = []
+        def dataConstraints = [:]
         filters.each { Map filter ->
-            log.info "creating filter of type ${filter.type}..."
+            log.debug "creating filter of type ${filter.type}..."
             def type = filter.type
             def data = filter.findAll { it.key != 'type' && it.key != 'id' }
-            log.info "  data: ${data}"
-            def constraint = dataTypeResource.createDataConstraint(data, type)
-            log.info "  constraint: ${constraint}"
-            dataConstraints << constraint
+            log.debug "  data: ${data}"
+            if (!(type in dataConstraints)) {
+                dataConstraints[type] = []
+            }
+            dataConstraints[type] << data
```

Comment:
Looks like you're using a default map. Use "def dataConstraints = [:].withDefault { k -> [] }".